File: benchmarks/node/05_spectral.mjs
function spectral(n) {
    let u = new Float64Array(n).fill(1.0);
    let v = new Float64Array(n);
    for (let iter = 0; iter < 10; iter++) {
        for (let i = 0; i < n; i++) {
            let s = 0;
            for (let j = 0; j < n; j++) s += u[j] / ((i+j)*(i+j+1)/2 + i + 1);
            v[i] = s;
        }
        for (let i = 0; i < n; i++) {
            let s = 0;
            for (let j = 0; j < n; j++) s += v[j] / ((i+j)*(i+j+1)/2 + i + 1);
            u[i] = s;
        }
    }
    console.log(u[0]);
}
spectral(5000);

File: benchmarks/python/05_spectral.py
def spectral(n):
    u = [1.0] * n
    v = [0.0] * n
    for _ in range(10):
        for i in range(n):
            s = 0.0
            for j in range(n):
                s += u[j] / ((i + j) * (i + j + 1) / 2 + i + 1)
            v[i] = s
        for i in range(n):
            s = 0.0
            for j in range(n):
                s += v[j] / ((i + j) * (i + j + 1) / 2 + i + 1)
            u[i] = s
    print(u[0])
spectral(5000)
